### PR TITLE
MINOR: fix error message

### PR DESCRIPTION
**Pull request by mjsax**
Got the accumulator lambda wrong in #9688 

Call for review @ableegoldman

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java [hunk 1]
```diff
@@ -1045,8 +1045,8 @@ private void checkResultPerKey(final List<KeyValue<Long, Long>> result,
                 assertThat(getAllRecordPerKey(key, result), equalTo(getAllRecordPerKey(key, expectedResult)));
             } catch (final AssertionError error) {
                 throw new AssertionError(
-                    "expected result: " + expectedResult.stream().map(KeyValue::toString).reduce("", (kv, str) -> str.isEmpty() ? kv : str + ", " + kv) +
-                    "\nreceived records: " + result.stream().map(KeyValue::toString).reduce("", (kv, str) -> str.isEmpty() ? kv : str + ", " + kv),
+                    "expected result: " + expectedResult.stream().map(KeyValue::toString).reduce("", (str, kv) -> str.isEmpty() ? kv : str + ", " + kv) +
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java]: How about using ```.collect(Collectors.joining(",")```?

**Review by ableegoldman — APPROVED** [review]

File: streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java (hunk 1 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java]: Still a noob using the JavaStream API... Does this only use the separator if there are more than one element?

Comment (chia7712) [reply streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java]: > Does this only use the separator if there are more than one element?

yep

**Review by chia7712 — APPROVED** [review]
@mjsax +1 to latest commit :)

**Comment by mjsax** [comment]
Merged to trunk and cherry-picked to `2.7` branch.

**Pull request merged**